### PR TITLE
Tensor init clean

### PR DESCRIPTION
**Pull request by Liamdoult**

### CODE REVIEW

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
 from functools import partialmethod
 from inspect import signature
+import logging
+
 import numpy as np
 
 # **** start with two base classes ****
 
 class Tensor:
+  # internal variables used for autograd graph construction
```
Comment (geohot) [inline tinygrad/tensor.py:11]: I don't think this is right, this shouldn't be class global

File: tinygrad/tensor.py [hunk 2]
```diff
@@ -1,27 +1,28 @@
 # inspired by https://github.com/karpathy/micrograd/blob/master/micrograd/engine.py
 from functools import partialmethod
 from inspect import signature
+import logging
```
Comment (geohot) [inline tinygrad/tensor.py:4]: Hate this package

File: tinygrad/tensor.py [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
   def __init__(self, data):
-    #print(type(data), data)
-    if type(data) == list:
+    if isinstance(data, list):
       data = np.array(data, dtype=np.float32)
-    if type(data) != np.ndarray:
-      print("error constructing tensor with %r" % data)
-      assert(False)
+    elif not isinstance(data, np.ndarray):
+      raise TypeError("Error constructing tensor with %r" % data)
```
Comment (geohot) [inline tinygrad/tensor.py:18]: This change I would merge

**Comment by Liamdoult** [comment]
shitted

**Pull request closed without merging**